app/full_text_search.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import re
+
+from google.appengine.api import search
+
+import model
+
+# The index name for full text search
+PERSON_FULLTEXT_INDEX_NAME = 'person_information'
+
+def search_with_index(repo, query_obj, max_results):
```

Comment:
Nit: You may just name this method "search" instead of "search_with_index" because it's obvious that it uses an index.